catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/CswTransformProvider.java

```diff
@@ -147,23 +147,29 @@ private void writeXml(BinaryContent content, HierarchicalStreamWriter writer) {
      */
     @Override
     public Object unmarshal(HierarchicalStreamReader reader, UnmarshallingContext context) {
-        Object arg = context.get(CswConstants.OUTPUT_SCHEMA_PARAMETER);
+        Object outputSchema = context.get(CswConstants.OUTPUT_SCHEMA_PARAMETER);
+        Object typeName = context.get(CswConstants.TYPE_NAME_PARAMETER);
         InputTransformer transformer = null;
-        if (arg == null || CswConstants.CSW_OUTPUT_SCHEMA.equals((String) arg)) {
+        if (StringUtils.equals(CswConstants.CSW_OUTPUT_SCHEMA, (String) outputSchema) || StringUtils
```

Comment:
It'd be marginally more efficient to switch the test ordering here and do your test for the two nulls first.